File: skill-capture-backend/src/models/employee.rs
use rocket::serde::{self, Serialize, Deserialize};
use std::time::SystemTime;

use super::skill::{ Skill };

use crate::schema::employee;

#[derive(Insertable, Identifiable)]
#[table_name = "employee"]
#[primary_key("id")]
pub struct AddEmployee {
    // id: i32,
    pub first_name: String,
    pub last_name: String,
    pub title: String
    // created_at: SystemTime,
    // updated_at: SystemTime,
}
#[derive(Queryable, Identifiable,Debug, Serialize, Deserialize)]
#[serde(crate="self::serde")]
#[table_name = "employee"]
#[primary_key("id")]
pub struct Employee {
    id: i32,
    pub first_name: String,
    pub last_name: String,
    pub title: String,
    created_at: SystemTime,
    updated_at: SystemTime
}

File: skill-capture-frontend/src/models/Employee.ts
export type Employee = {
    first_name: string;
    last_name: string;
    title: string;
    created_at: string;
    updated_at: string;
}

type mkEmployeeT = (
    first_name: string,
    last_name: string,
    title: string,
    created_at: string,
    updated_at: string) => Employee;
export const mkEmployee: mkEmployeeT = (first_name, last_name, title, created_at, updated_at) => ({
    first_name,
    last_name,
    title,
    created_at,
    updated_at,
});